### DISCUSSION TITLE
Unable to add more than 20 Questions for a Question set.

### DISCUSSION THREAD

**Discussion by ramkumar-ns-dev** [Sunbird-coKreat/Community#102] [General Discussions]
Prerequisites
I have searched for existing issues that already report this problem, without success.

Current Behaviour
I'm testing complete flow of Observation with rubrics using dock dev environment. While i'm testing i added 20 Questions to a Questionset, But while adding 21 question i'm getting the following Bad Request Error:
`"Request contains more than the permissible limit of identifiers: 21. Permissible limit is: 20"`

Expected Behaviour
while adding the Questions to Question-set user might need more than 20 Questions for a Question set. So expecting to added more than 20 Questions

Steps to Reproduce
1. Create a Question set in Observation with rubrics
2. Create Questions under any criteria.
3. after adding 20 Questions, while adding 21 Question will throw following error
`{
    "resmsgid": "842f42cf-e4ca-40e7-a7ff-f904892c40c3",
    "msgid": null,
    "err": "ERR_BAD_REQUEST",
    "status": "failed",
    "errmsg": "Request contains more than the permissible limit of identifiers: 21. Permissible limit is: 20"
}`

API Endpoints used in create/upload flow
    1. https://dock.sunbirded.org/action/question/v2/list

can anybody help on this.
@krgauraw @rajnishdargan

cc: @kiranharidas187

**Comment by krgauraw** [comment]
@ramkumar-pacewisdom this error is because of below configuration , where default is 20 question. 
question.list.limit={{ question_list_api_request_limit | default('20') }}

if you need more than 20, please request devops team to override question_list_api_request_limit variable with required value and re-deploy assessment micro-service.
cc: @kiranharidas187 @vinukumar-vs

**Reply by ramkumar-ns-dev** [reply]
@krgauraw But we can't sure about the max limitation of the Questions, Because in Observation with rubrics there will be multiple criteria and Questions. Is there any other way to get Questions details without certain limitation ? 
cc :  @kiranharidas187

**Reply by krgauraw** [reply]
Hi @ramkumar-pacewisdom we can't keep the configuration open or set a very high number because it may cause performance issue for other api's. We will have to do few rounds of performance testing in order to get a specific number of questions we can support in single request. we haven't done that yet.

So i would suggest you to go with 40 as configuration value as of now and lets discuss over call for observation with rubrics use case.

**Reply by ramkumar-ns-dev** [reply]
@krgauraw Thanks for the response. we'll discuss on that over a call.
